### PR TITLE
end to end guide for kubeflow and kfserving in aws

### LINKED ISSUE #851: AWS: Document manual steps for ASG/EC2 deletion
Kubeflow on AWS: After deleting EKS cluster and removing the EKS cluster in the AWS, the instances are still getting created, how do we stop the same?

### LINKED ISSUE #1435: AWS: Authentication and TLS Support
https://www.kubeflow.org/docs/aws/authentication/

It would be very useful if there was a more comprehensive guide / some troubleshooting assistance for this. Setting up authenticated access is pretty far from trivial.

I have followed the steps above, but am unable to reach the authetication screen. This is very hard to troubleshoot, since the problem could be in any one of

   1. Cognito user pool
   2. The Route 53 CNAME definition
   3. The Istio YAML definition
   4. AWS Certificate management

There are also a few inconsistencies on the page that are either mistakes or otherwise need to be fully explained:

- The example shows registering a custom domain (`www.shanjiaxin.com` in this case), but then in the Amazon Cognito app client settings, this is entered as `www.shanjiaxin.com/oauth2/idresponse`. Why isn't it just `www.shanjiaxin.com`? This should be explained!

- Under Cognito Domain Name the value `kubeflow-testing` is entered. In the example YAML snippet below you have `cognitoUserPoolDomain: your-user-pool`. Is far as I can tell, this should be `cognitoUserPoolDomain: kubeflow-testing` to be consistent with the rest of the example.

- For adding a CNAME under Points-To it appears to point to the ALB endpoint. It should be explained how this endpoint can be found and if it should be typed in exactly as it is. Since most users will probably use Route 53 for this, a guide that shows the steps for Route 53 would be sensible

### PR DESCRIPTION
**Pull request by theofpa**
fixes #851, #1435 and #1541

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1661)

### CODE REVIEW

**Comment by theofpa** [comment]
@Jeffwan could you please support with the review?

**Comment by Jeffwan** [comment]
Thanks! This is a great doc showing e2e examples from users view.

File: content/docs/aws/aws-e2e.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+## Deploy Kubeflow
+
+In this section you will prepare the ecosystem required by kubeflow, and you will configure the kfctl.yaml file with the custom information for your environment.
+
+### Cognito and certificates
+
+#### Route53
+
+Register or delegate a domain in a Route53 hosted zone, in our example we have delegated the subdomain platform.domain.com.
```
Comment (Jeffwan) [inline content/docs/aws/aws-e2e.md]: I am not sure if this is clear enough for most of the users.

File: content/docs/aws/aws-e2e.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+#### Certificate Manager
+
+Create two certificates in Certificate Manager for `*.platform.domain.com`, one in N.Virginia and one in the region of your choice. That is because Cognito [requires](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html) a certificate in N.Virginia in order to have a custom domain for a user pool. The second is required by the ingress-gateway in case the platform does not run in N.Virginia, in our example Dublin.
+
+#### Cognito
+
+Create a user pool in Cognito. Type a pool name and choose `Review defaults` and `Create pool`.
+
+Create some users in `Users and groups`, these are the users who will login to the central dashboard.
```
Comment (Jeffwan) [inline content/docs/aws/aws-e2e.md]: Later on, we can add more docs to talk about third party user pool federation. This is good enough now.

File: content/docs/aws/aws-e2e.md [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+
+In the `App client settings` select `Authorization code grant` flow and `email`, `openid`, `aws.cognito.signin.user.admin` and `profile` scopes.
+
+In the `Domain name` choose `Use your domain`, type `auth.platform.domain.com` and select the `*.platform.domain.com` AWS managed certificate you've created in N.Virginia. When it's created, it will return the `Alias target` cloudfront address for which you need to create a CNAME `auth.platform.domain.com` in the hosted zone.
+
+Take note of the following 5 values:
+* The ARN of the certificate from the Certificate Manager of N.Virginia (<certArn>).
+* The Pool ARN (<cognitoUserPoolArn>) of the user pool found in Cognito general settings.
+* The App client id (<cognitoAppClientId>), found in Cognito App clients.
+* The `auth.platform.domain.com` as the <cognitoUserPoolDomain>.
```
Comment (Jeffwan) [inline content/docs/aws/aws-e2e.md]: I think this address the some community questions using custom domain. Thanks.

**Comment by Jeffwan** [comment]
/lgtm

looks good to me on the technical side

File: content/docs/aws/aws-e2e.md (hunk 1 above)
Comment (theofpa) [reply content/docs/aws/aws-e2e.md]: addressed in commit 9ffa0d1

**Comment by Jeffwan** [comment]
/lgtm

@sarahmaddox

**Comment by sarahmaddox** [comment]
Preview:
* https://deploy-preview-1661--competent-brattain-de2d6d.netlify.com/docs/aws/aws-e2e/
* https://deploy-preview-1661--competent-brattain-de2d6d.netlify.com/docs/aws/deploy/uninstall-kubeflow/

**Comment by sarahmaddox** [comment]
Thanks!
/lgtm
/approve

**Pull request merged**